app/controllers/receive_package.js

```diff
@@ -28,180 +30,182 @@ export default Ember.Controller.extend({
-        inventoryNumber: pkg.get("inventoryNumber"),
         notes: pkg.get("notes")
       };
-    },
-    set: function(key, value) {
-      return {
-        quantity: value.get("quantity"),
-        length: value.get("length"),
-        width: value.get("width"),
-        height: value.get("height"),
-        inventoryNumber: value.get("inventoryNumber"),
-        notes: value.get("notes")
-      };
     }
   }),
 
-  hasErrors: Ember.computed('invalidQuantity', 'invalidInventoryNo', 'invalidDescription', 'invalidLocation', 'watchErrors', {
-    get: function() {
-      return this.get("invalidQuantity") || this.get("invalidInventoryNo") || this.get("invalidDescription") || this.get("invalidLocation");
-    },
-    set: function(key, value) {
-      return value;
-    }
+  enableDisplayError() {
+    this.set("displayError", true);
+  },
+
+  isInvalidQuantity: Ember.computed("packageForm.quantity", function() {
+    return this.get("packageForm.quantity") < 1;
   }),
 
-  invalidQuantity: Ember.computed({
-    get: function() {
-      return this.get("package.quantity").length === 0;
-    },
-    set: function(key, value) {
-      return value;
-    }
+  isInvalidLocation: Ember.computed("locationId", function() {
+    return !this.get("locationId");
   }),
 
-  invalidLocation: Ember.computed('locationId', {
-    get: function() {
-      return this.get("locationId") === undefined;
-    },
-    set: function(key, value) {
-      return value;
-    }
+  isInvalidDescription: Ember.computed("packageForm.notes", function() {
+    return this.get("package.notes").length === 0;
   }),
 
-  invalidDescription: Ember.computed({
-    get: function() {
-      return this.get("package.notes").length === 0;
-    },
-    set: function(key, value) {
-      return value;
-    }
+  isInvalidInventoryNo: Ember.computed("inventoryNumber", function() {
+    return !this.verifyInventoryNumber(this.get("inventoryNumber"));
   }),
 
-  invalidInventoryNo: Ember.computed({
-    get: function() {
-      var isValid = this.verifyInventoryNumber(this.get("package.inventoryNumber"));
-      return isValid;
-    },
-    set: function(key, value) {
-      return value;
+  isPackageInvalid: Ember.computed(
+    "isInvalidQuantity",
+    "isInvalidInventoryNo",
+    "isInvalidDescription",
+    "isInvalidLocation",
+    function() {
+      return (
+        this.get("isInvalidQuantity") ||
+        this.get("isInvalidInventoryNo") ||
+        this.get("isInvalidDescription") ||
+        this.get("isInvalidLocation")
+      );
     }
-  }),
+  ),
+
+  receivePackageParams() {
+    const pkgData = this.get("packageForm");
+    const inventoryNumber = this.get("inventoryNumber");
+    let pkg = this.get("package");
+    const locationId = this.get("locationId.id") || this.get("locationId");
+    if (locationId) {
+      var location = this.get("store").peekRecord("location", locationId);
+      pkg.set("location", location);
+    }
+    pkg.set("state", "received");
```

Comment:
ember objects have a `setProperties` method as well, which can be used to set multiple properties.
I don't really think one way is better than the other but just thought I'd mention it : )